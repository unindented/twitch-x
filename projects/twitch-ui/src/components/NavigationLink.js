import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import NavigationItem from './NavigationItem'

const NavigationLinkRoot = styled.li`
  display: block;
`

const NavigationLinkReal = styled.a`
  display: block;
  color: ${props => props.theme.colors.white};
  text-decoration: none;

  &:hover {
    background-color: ${props => props.theme.colors.tertiary};
  }
`

export default function NavigationLink (props) {
  const {href, icon, iconAlt, label} = props

  return (
    <NavigationLinkRoot>
      <NavigationLinkReal
        href={href}
      >
        <NavigationItem
          icon={icon}
          iconAlt={iconAlt}
          label={label}
        />
      </NavigationLinkReal>
    </NavigationLinkRoot>
  )
}

NavigationLink.propTypes = {
  href: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired,
  iconAlt: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired
}
