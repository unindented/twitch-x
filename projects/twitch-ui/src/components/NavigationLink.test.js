import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'
import NavigationLink from './NavigationLink'

describe('NavigationLink', () => {
  let tree

  beforeEach(() => {
    const renderer = new ShallowRenderer()
    renderer.render(<NavigationLink
      href='/'
      icon='home'
      iconAlt=''
      label='Home'
    />)
    tree = renderer.getRenderOutput()
  })

  it('renders', () => {
    expect(tree).toMatchSnapshot()
  })
})
