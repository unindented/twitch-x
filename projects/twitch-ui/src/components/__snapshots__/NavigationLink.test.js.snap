// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationLink renders 1`] = `
<NavigationLink__NavigationLinkRoot>
  <NavigationLink__NavigationLinkReal
    href="/"
  >
    <NavigationItem
      icon="home"
      iconAlt=""
      label="Home"
    />
  </NavigationLink__NavigationLinkReal>
</NavigationLink__NavigationLinkRoot>
`;
